src/components/Cashier/Deposit.tsx

```diff
@@ -44,30 +45,46 @@ const DepositAddressContainer = styled.div`
   padding: 0.5rem;
 `;
 
-const Deposit: React.FunctionComponent<IProps> = ({ state, dispatch }) => {
+const Deposit: React.FunctionComponent<IProps> = ({
+  state,
+  closeCashierModal
+}) => {
   const { balance, depositAddress } = state;
 
   const [isAddressCopied, setIsAddressCopied] = useState(false);
+  const [isDepositAddressValid, setIsDepositAddressValid] = useState(false);
 
-  const handleSubmit = () => (): void => {
-    updateStateValue("isCashierOpen", false, dispatch);
-  };
+  // Validate if the deposit address is correct
+  useEffect(() => {
+    setIsDepositAddressValid(isValidAddress(depositAddress));
+  }, [balance]);
 
+  // Copy the address to the clipboard and hide the tooltip when clicked
   const copyToClipBoard = () => (): void => {
     navigator.clipboard.writeText(depositAddress);
     setIsAddressCopied(true);
     ReactTooltip.hide();
   };
 
+  // Set the cursor style based on whether the deposit address is valid
+  const cursorStyle = css`
+    cursor: ${isDepositAddressValid ? "pointer" : "not-allowed"};
+  `;
+
   return (
     <section>
-      <Balance>Available CHIPS: {balanceWithDecimals(balance)}</Balance>
+      <Balance data-test="balance-cashier-deposit">
+        Available CHIPS: {balanceWithDecimals(balance)}
+      </Balance>
       <AddressLabel>Your CHIPS deposit address:</AddressLabel>
       <DepositAddressContainer
         data-tip={isAddressCopied ? "Copied!" : "Copy to Clipboard"}
-        onClick={copyToClipBoard()}
+        onClick={isDepositAddressValid ? copyToClipBoard() : undefined}
```

Comment:
Thanks. The reason I'm not using `isDepositAddressValid && copyToClipBoard()` is because React is complaining about it in form of a console error - it  asks me to use a ternary instead.